packages/core/test/interpreter.test.ts

```diff
@@ -118,6 +118,56 @@ describe('interpreter', () => {
         done();
       }, 100);
     });
+
+    // https://github.com/davidkpiano/xstate/issues/1174
+    it('executes actions from a restored initial state', (done) => {
```

Comment:
nit: I wouldn't call it initial state - but rather just restored or rehydrated. The initial state (the name) is already used in other places such as here:
https://github.com/davidkpiano/xstate/blob/f685e674ac9d57f832df3f932af7de848f1e0d7f/packages/core/src/interpreter.ts#L225
The accepted parameter is called `initialState` here though:
https://github.com/davidkpiano/xstate/blob/f685e674ac9d57f832df3f932af7de848f1e0d7f/packages/core/src/interpreter.ts#L462-L466
so it follows that OK, although I find this parameter name confusing in the very same way.